core/primitives/src/optimistic_block.rs

```diff
@@ -0,0 +1,82 @@
+use borsh::{BorshDeserialize, BorshSerialize};
+use near_crypto::Signature;
+
+use crate::block::BlockHeader;
+use crate::hash::{hash, CryptoHash};
+use crate::merkle::combine_hash;
+use crate::stateless_validation::SignatureDifferentiator;
+use crate::types::BlockHeight;
+
+#[derive(BorshSerialize, BorshDeserialize, Clone, Debug, Eq, PartialEq)]
+pub struct OptimisticBlockInner {
+    signature_differentiator: SignatureDifferentiator,
```

Comment:
For all structures where it is currently used, it is in the end of structure, so here we should also put it to the end. It makes reasoning about difference of signed bytes much simpler.